peer/test_utils.go

```diff
@@ -53,14 +53,19 @@ var (
 // the channels set up.
 var noUpdate = func(a, b *channeldb.OpenChannel) {}
 
+type testPeerChanEnv struct {
+	alicePeer  *Brontide
+	bobChannel *lnwallet.LightningChannel
+}
+
 // createTestPeerWithChannel creates a channel between two nodes, and returns a
 // peer for one of the nodes, together with the channel seen from both nodes.
 // It takes an updateChan function which can be used to modify the default
 // values on the channel states for each peer.
 func createTestPeerWithChannel(t *testing.T, notifier chainntnfs.ChainNotifier,
 	publTx chan *wire.MsgTx, updateChan func(a, b *channeldb.OpenChannel),
 	mockSwitch *mockMessageSwitch) (
-	*Brontide, *lnwallet.LightningChannel, error) {
+	*testPeerChanEnv, error) {
```

Comment:
So that one can easily return nil? I also read that working with pointers is more memory efficient instead of copying struct into different memory addresses. It is also in line with what I have seen in the codebase

Reply:
In this case you can leave it. Discussions of efficiency are highly nuanced, and what you read isn't wrong, but it isn't right either. Also we are moving away from excessive pointer usage throughout the codebase.